### PR TITLE
feat: added cart management

### PR DESCRIPTION
**Pull request by nerdyspook**
**Added Cart Management:**
- When Logged in:
-> `ADD TO CART` will add the product to the cart.
-> if the product is already present `ADD TO CART` will increase the quantity.
-> `❤` will add the product to the wishlist and remove the product from the wishlist.
-> colour of the `❤` indicates whether the product is on the wishlist or not. Red indicates that the product is on the wishlist, and Gray indicates that the product is not on the wishlist.

- When NOT Logged in:
-> `ADD TO CART` will redirect to the login page.
-> `❤` will redirect to the login page.

- In Cart Page:
-> can increase the quantity of any product by using the `+` button and decrease the quantity of any product by using the `-` button.
-> the total price will automatically increase or decrease according to the quantity of products in the cart.
-> can remove any product from the cart using the `remove` button.

### CODE REVIEW

File: src/components/CartProduct/CartProduct.jsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import QuantityCounter from "../QuantityCounter/QuantityCounter";
+import { useCart } from "../../contexts/CartContext";
+import { addWishlist } from "../../utilities/add-wishlist";
+import { removeCart } from "../../utilities/remove-cart";
+
+const CartProduct = ({ product }) => {
+    const { stateCart, dispatchCart } = useCart();
+    const { myWishlist } = stateCart;
+    const { _id, id, img, title, details, price, qty } = product;
+    console.log(qty, id, product);
```
Comment (amarphule) [inline src/components/CartProduct/CartProduct.jsx]: you can remove a console

File: src/components/QuantityCounter/QuantityCounter.jsx [hunk 2]
```diff
@@ -1,21 +1,47 @@
...
+            <div className="quantity__count">
+                {/* {count} */}
+                {qty}
+            </div>
             <button
                 className="increment"
-                onClick={() => setCount((prevCount) => prevCount + 1)}
+                // onClick={() => setCount((prevCount) => prevCount + 1)}
+                onClick={() => {
+                    console.log(product);
```
Comment (amarphule) [inline src/components/QuantityCounter/QuantityCounter.jsx]: can remove console

File: src/pages/Wishlist/Wishlist.jsx [hunk 3]
```diff
@@ -1,35 +1,24 @@
-import React, { useEffect, useState } from "react";
-import { products } from "../../fakeData";
+// import React, { useEffect, useState } from "react";
```
Comment (amarphule) [inline src/pages/Wishlist/Wishlist.jsx]: if not using imported hooks then instead of commenting you can remove it

**Review by amarphule — APPROVED** [review]
Very well structured code, nice work.

File: src/pages/Cart/Cart.jsx [hunk 4]
```diff
@@ -1,81 +1,51 @@
...
-                        <img src={product.img} alt={product.title} />
-                        <div className="card-details">
-                            <div className="heading">
-                                <h4>
-                                    {product.details}{" "}
-                                    <span>{`₹${product.price}`}</span>
-                                </h4>
-                                <h6>{product.title}</h6>
-                            </div>
+    console.log(myCart);
```
Comment (prathmeshjagtap) [inline src/pages/Cart/Cart.jsx]: remove console.log

File: src/utilities/add-wishlist.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import axios from "axios";
+
+export const addWishlist = async (product, dispatchCart) => {
+    console.log(product);
+    const newToken = localStorage.getItem("token");
```
Comment (prathmeshjagtap) [inline src/utilities/add-wishlist.js]: remove console.log

File: src/utilities/add-wishlist.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    try {
+        const response = await axios({
+            method: "POST",
+            url: "/api/user/wishlist",
+            data: { product: product },
+            headers: {
+                authorization: newToken,
+            },
+        });
+        console.log(response);
```
Comment (prathmeshjagtap) [inline src/utilities/add-wishlist.js]: same here

File: src/utilities/quantity-cart.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+export const quantityCart = async (
+    id,
+    dispatchCart,
+    quantityIncrementDecrement,
+    product
+) => {
+    const newToken = localStorage.getItem("token");
+    console.log(id, quantityIncrementDecrement, product);
+
```
Comment (prathmeshjagtap) [inline src/utilities/quantity-cart.js]: same here

File: src/utilities/quantity-cart.js [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+                    type: quantityIncrementDecrement,
+                },
+            },
+
+            headers: {
+                authorization: newToken,
+            },
+        });
+        console.log(response);
+
```
Comment (prathmeshjagtap) [inline src/utilities/quantity-cart.js]: same here

**Review by prathmeshjagtap — APPROVED** [review]
Everything else is good and Well structured

**Pull request merged**